File: aumms/public/js/purchase_invoice.js
frappe.ui.form.on('Purchase Invoice', {

 });

frappe.ui.form.on('Purchase Invoice Item', {
  item_type(frm, cdt, cdn) {
    let child = locals[cdt][cdn]
    if (child.item_type) {
      frappe.call ({
        method: 'aumms.aumms.doc_events.purchase_invoice.check_is_purity_item',
        args: {
          'item_type': child.item_type
        },
        callback: function(r) {
          if (r.message) {
            child.is_purity_item = r.message
          }
          else {
            child.is_purity_item = 0
          }
        }
      })
    }
  },
  items_add(frm, cdt, cdn){
    let child = locals[cdt][cdn]
    //Checking the keep metal ledger
    if(frm.doc.keep_metal_ledger){
      //set value to the field is keep_metal_ledger as 1
      frappe.model.set_value(child.doctype, child.name, 'keep_metal_ledger', 1)
    }
  }

});

File: aumms/aumms/doc_events/purchase_invoice.py
import frappe
from frappe import _
from erpnext.stock.doctype.purchase_receipt.purchase_receipt import make_purchase_invoice


@frappe.whitelist()
def check_is_purity_item(item_type):
    ''' Method for fetching items with is_purity_item as 1 '''
    is_purity = ''
    if frappe.db.exists('Item Type', {'name': item_type, 'is_purity_item': 1}):
        is_purity = frappe.db.get_value('Item Type', {'name': item_type , 'is_purity_item': 1}, 'is_purity_item')
    return is_purity
